backend/bot/index.js

```diff
@@ -1,52 +1,46 @@
 const SlackBot = require('slackbots');
 
-class FoosBot extends SlackBot {
+class FoosBot {
+  constructor(channelId, slackbot) {
+    this.channelId = channelId
+    this.slackbot = slackbot
+  }
+
   /**
    * Sets the purpose for a private channel
-   * @param {string} id - channel ID
+
    * @param {string} purpose
    * @returns {vow.Promise}
    */
-  setGroupPurpose(id, purpose) {
-    return this._api('groups.setPurpose', { channel: id, purpose });
+  setGroupPurpose(purpose) {
+    return this.slackbot._api('groups.setPurpose', { channel: this.channelId, purpose });
   }
 
-}
-
-const bot = new FoosBot({
-  token: process.env.FOOSBOT_TOKEN
-});
-
-let channel;
-
-bot.on('start', async () => {
-  console.log('Foosbot started');
-  channel = await bot.getGroup(process.env.FOOS_CHANNEL_NAME);
-  if (!channel) {
-    throw `Channel '${process.env.FOOS_CHANNEL_NAME}' not found!`
+  postMessage(message, isAsUser = true) {
+    return this.slackbot.postMessage(this.channelId, message, { as_user: isAsUser })
   }
-});
-
-
-const ratingComparator = (a, b) => b.rating - a.rating
-
-const reportMatchOnSlack = async (match, oldUsers, newUsers) => {
-  await postMatchResult(match)
 
-  const oldRankings = oldUsers.sort(ratingComparator)
-  const newRankings = newUsers.sort(ratingComparator)
-
-  await postRankingChangeMessage(oldRankings, newRankings)
-
-  const leaderboardSize = 5
-  const shouldUpdatePurpose = hasLeaderboardChanged(leaderboardSize, oldRankings, newRankings)
-
-  if (shouldUpdatePurpose) {
-    await updatePurpose(newRankings.slice(0, leaderboardSize))
+  reportMatchOnSlack = async (match, oldUsers, newUsers) => {
+    let matchResultMessage = createMatchResultMessage(match)
+  
+    const oldRankings = oldUsers.sort(ratingComparator)
+    const newRankings = newUsers.sort(ratingComparator)
+  
+    const rankingChangeMessage = createRankingChangeMessage(oldRankings, newRankings)
+    await this.postMessage(`${matchResultMessage}\n${rankingChangeMessage}`);
```

Comment:
Please remove the semicolon, it is the prevailing style not to use them.

Reply:
sure, like 70% of commands in the file are semicolon free ;-)) 
fixed, + removed semicolons in the rest of the file